### PR TITLE
kubeadm: fix conversion macros and add kubeadm to round trip testing

### PR DESCRIPTION
**Pull request by mikedanese**
Tests are probably broken but I'll fix. @jbeda this probably fixes your change unless we decide we need generated deep copies or conversions.

@kubernetes/sig-cluster-lifecycle

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/34555)

### CODE REVIEW

**Comment by jbeda** [comment]
I'm kind of a noob on this stuff but LGTM. I'll close out #34497 as redundant.

File: pkg/api/testapi/testapi.go [hunk 1]
```diff
@@ -257,6 +258,15 @@ func init() {
 			externalTypes:        api.Scheme.KnownTypes(externalGroupVersion),
 		}
 	}
+	if _, ok := Groups[kubeadm.GroupName]; !ok {
```
Comment (krousey) [inline pkg/api/testapi/testapi.go]: This is ugly. I would prefer this and/or other packages providing APIs to use to do the testing without having to modify this package and register it globally. Way out of scope though.

**Comment by luxas** [comment]
`verify-boilerplate` and `hack/verify-golint` failed.

I don't either know much about this, but maybe @lavalamp has some comments?

File: cmd/kubeadm/app/apis/kubeadm/install/install.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+		&announced.GroupMetaFactoryArgs{
+			GroupName:                  kubeadm.GroupName,
+			VersionPreferenceOrder:     []string{v1alpha1.SchemeGroupVersion.Version},
+			ImportPrefix:               "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm",
+			AddInternalObjectsToScheme: kubeadm.AddToScheme,
+		},
+		announced.VersionToSchemeFunc{
+			v1alpha1.SchemeGroupVersion.Version: v1alpha1.AddToScheme,
+		},
+	).Announce().RegisterAndEnable(); err != nil {
```
Comment (krousey) [inline cmd/kubeadm/app/apis/kubeadm/install/install.go]: This machinery is much different than what I remember. What is the difference between announce, register, and enable? And why to we need each?

Comment (mikedanese) [inline cmd/kubeadm/app/apis/kubeadm/install/install.go]: No idea. Maybe @lavalamp or @caesarxuchao can say. I need this apigroup to be available from `registered.GroupOrDie` so I can add it to the testapi package and thus round trip testing.

Comment (lavalamp) [inline cmd/kubeadm/app/apis/kubeadm/install/install.go]: Announce and register are really the same concept, but my refactoring work is half done.

File: pkg/api/testapi/testapi.go (hunk 1 above)
Comment (lavalamp) [inline pkg/api/testapi/testapi.go]: This whole package should not exist, we should reuse the manager in apimachinery/registered. It's a big change though. @mml This is one of the refactoring changes I want to make.

**Comment by lavalamp** [comment]
changes lgtm if you get the tests passing.

**Comment by mikedanese** [comment]
The tests are failing only for the protobuf round trip which we haven't generated. I wish we didn't need proto for a file based config api group. I found numerous rough edges today debugging this.

**Comment by errordeveloper** [comment]
@mikedanese:
```
Boilerplate header is wrong for: /go/src/k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/install/install.go
FAILED   hack/make-rules/../../hack/verify-boilerplate.sh	2s
```

**Comment by errordeveloper** [comment]
Can we get this and #34501 merged today?

File: cmd/kubeadm/app/apis/kubeadm/v1alpha1/types.go [hunk 3]
```diff
@@ -23,6 +23,7 @@ type MasterConfiguration struct {
 
 	Secrets           Secrets    `json:"secrets"`
 	API               API        `json:"api"`
+	Etcd              Etcd       `json:"etcd"`
```
Comment (errordeveloper) [inline cmd/kubeadm/app/apis/kubeadm/v1alpha1/types.go]: I was wondering why these were missing 👍

**Comment by krousey** [comment]
The changes lgtm to me so long as the tests pass.

**Comment by mikedanese** [comment]
@krousey please take one more look.

**Review by krousey — APPROVED** [review]
Stop pointing out how gross this is.

/lgtm

**Comment by mikedanese** [comment]
I don't get this:

```
Errors from golint:
cmd/genyaml/gen_kubectl_yaml.go:86:2: can probably use "var result []cmdOption" instead
cmd/genyaml/gen_kubectl_yaml.go:134:3: can probably use "var result []string" instead
plugin/pkg/auth/authenticator/request/x509/x509_test.go:768:2: can probably use "var certs []*x509.Certificate" instead

Please fix the above errors. You can test via "golint" and commit the result.
```

**Comment by mikedanese** [comment]
golint does maintain compatibility between releases. I don't think this is a good check.

**Comment by errordeveloper** [comment]
> golint does maintain compatibility between releases. I don't think this is a good check.

Makes sense.

**Comment by errordeveloper** [comment]
@mikedanese something weird is going on with @k8s-merge-robot, seems like test are passing but it got confused...

**Comment by errordeveloper** [comment]
I am suspecting/guessing that just reporting back to Github is broken, Jenkins was actually green for last 'Bootstrap GCE e2e' but it wasn't reflected here.

**Comment by spxtr** [comment]
https://github.com/kubernetes/test-infra/issues/812

**Comment by errordeveloper** [comment]
@k8s-bot test this

**Pull request merged**